### PR TITLE
cleanup test harness code

### PR DESCRIPTION
**Pull request by vladima**
1. use `Path` to store files in harness compiler host
2. remove unused code, convert `HarnessCompiler` to module as it is does not use any state

### CODE REVIEW

**Comment by mhegazy** [comment]
thanks!

**Comment by mhegazy** [comment]
:+1:

File: src/harness/harness.ts [hunk 1]
```diff
@@ -1034,95 +1027,74 @@ namespace Harness {
...
-                setCompilerOptionsFromHarnessSetting(this.settings, options);
+                let useCaseSensitiveFileNames = Harness.IO.useCaseSensitiveFileNames();
+                if (harnessSettings) {
+                    setCompilerOptionsFromHarnessSetting(harnessSettings, options);
+                }
+                if (options.useCaseSensitiveFileNames !== undefined) {
+                    useCaseSensitiveFileNames = options.useCaseSensitiveFileNames;
+                }
+                const getCanonicalFileName = ts.createGetCanonicalFileName(useCaseSensitiveFileNames);
+                const inputFilesWiithPath = inputFiles.map(f => {
```
Comment (weswigham) [inline src/harness/harness.ts]: `inputFilesWiithPath`

Comment (vladima) [inline src/harness/harness.ts]: THANKS

**Pull request merged**